core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -4131,6 +4131,22 @@ private SqlDialect nonOrdinalDialect() {
         .withClickHouse().ok(expectedClickHouse);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-7067">[CALCITE-7067]
+   * Maximum precision of unsigned bigint type in MysqlSqlDialect should be 20</a>. */
+  @Test void testCastToUBigInt() {
+    String query = "select cast(18446744073709551615 as bigint unsigned) from \"product\"";
+    final String expectedMysql = "SELECT CAST(18446744073709551615 AS BIGINT UNSIGNED)\n"
+        + "FROM `foodmart`.`product`";
+    final String errMsg = "org.apache.calcite.runtime.CalciteContextException: "
+        + "From line 1, column 13 to line 1, column 32: "
```

Comment:
I hope that the error message here only contains the phrase "Numeric literal '18446744073709551615' is out of range". This might be another issue that needs to be addressed. LGTM NOW.

Reply:
> I hope that the error message here only contains the phrase "Numeric literal '18446744073709551615' is out of range". This might be another issue that needs to be addressed. LGTM NOW.

Instead of RuntimException, this is CalciteContextException with line and column info.